server/src/main/java/org/eclipse/lsp/cobol/service/delegates/validations/CobolLanguageEngineFacade.java

```diff
@@ -110,13 +110,13 @@ private AnalysisResult toAnalysisResult(ResultWithErrors<SemanticContext> result
   }
 
   private List<CodeBlockDefinitionNode> collectCodeBlockNodes(
-      SemanticContext context, NodeType nodeType) {
-    return context
-        .getRootNode()
-        .getDepthFirstStream()
-        .filter(it -> it.getNodeType().equals(nodeType))
-        .map(CodeBlockDefinitionNode.class::cast)
-        .collect(toList());
+          SemanticContext context, NodeType nodeType) {
+    return Objects.isNull(context.getRootNode()) ? emptyList() : context
```

Comment:
Maybe it would be better to use Optional here?

Reply:
Yes, I will update.